### DISCUSSION TITLE
Why does Either.Match have the Left lambda on the right and the Right lambda on the left?

### DISCUSSION THREAD

**Discussion by MrYossu** [louthy/language-ext#1167] [General]
Been wondering this for ages, but never got around to ask.

If you look at a simple usage of `Either`, it could look like this...

```c#
SomeEitherProducingMethod()
  .Match(right => DoRightStuff(right), left => DoLeftStuff(left));
```

Stupid question, but why is the lambda that handles the case when `SomeEitherProducingMethod()` produces a `Left` on the right (ie second parameter to `Match`), and the lambda for `Right` on the left (ie first parameter)?

**Comment by louthy** [comment]
Just a legacy bad decision, but it can be made less problematic if you stack them:
```c#
SomeEitherProducingMethod()
  .Match(r => DoRightStuff(r), 
         l => DoLeftStuff(l));
```

Or use argument names, which is good practice generally, as it's much more declarative:
```c#
SomeEitherProducingMethod()
  .Match(Left: l => DoLeftStuff(l), Right: r => DoRightStuff(r));
```

**Reply by EivindAntonsen** [reply]
I'd argue it's only more declarative if the argument names are one-letter abbreviations. :P Otherwise it may just be verbosity without gain. "Left: l" could just have been "left" you know?

**Comment by MrYossu** [comment]
Interesting, I vaguely assumed you'd picked it up from Haskell or something.

Doesn't really worry me, as I rarely think of them as left and right, I usually think of them as the happy path and sad path (picked those terms  up from Scott Wlaschin), so it doesn't really matter which comes first.

Thanks for clearing it up.
